src/generate_conf.rs

```diff
@@ -85,9 +85,60 @@ fn generate_config(
         .ok_or_else(|| anyhow!("Invalid NM configuration"))?
         .to_owned();
 
+    populate_connection_ids(&mut interfaces, &config)?;
+    validate_connection_ids(&interfaces)?;
+
     Ok((interfaces, config))
 }
 
+fn validate_connection_ids(interfaces: &[Interface]) -> anyhow::Result<()> {
+    let empty_connection_ids: Vec<String> = interfaces
+        .iter()
+        .filter(|i| i.connection_ids.is_empty())
+        .map(|i| i.logical_name.to_owned())
+        .collect();
+
+    if !empty_connection_ids.is_empty() {
+        return Err(anyhow!(
+            "Detected interfaces without connection files: {}",
+            empty_connection_ids.join(", ")
+        ));
+    };
+
+    Ok(())
+}
+
+fn populate_connection_ids(
+    interfaces: &mut [Interface],
+    config: &NetworkConfig,
+) -> anyhow::Result<()> {
+    config
+        .iter()
+        .filter(|(filename, _)| filename != "lo.nmconnection")
```

Comment:
Would this cause an issue with `validate_connection_ids`? 👀

Reply:
No, the current implementation removes the `type=loopback` interfaces. So this code would fail for `lo.nmconnection` because it cannot find a matching interface definition.

This is however not perfect I realize now, there could be more loopback connections with each their own connection file.

Reply:
Precisely because we filter it out, I'd expect the `validate_connection_ids` to return an error and abort the generation in the cases where there are loopbacks.